cardano-slotting/src/Cardano/Slotting/Slot.hs

```diff
@@ -114,9 +118,28 @@ withOriginFromMaybe (Just t) = At t
 newtype EpochNo = EpochNo {unEpochNo :: Word64}
   deriving stock (Eq, Ord, Generic)
   deriving Show via Quiet EpochNo
-  deriving newtype (Enum, Num, Serialise, ToCBOR, FromCBOR, NoThunks, ToJSON, FromJSON, NFData)
+  deriving newtype (Enum, Serialise, ToCBOR, FromCBOR, NoThunks, ToJSON, FromJSON, NFData)
 
 newtype EpochSize = EpochSize {unEpochSize :: Word64}
   deriving stock (Eq, Ord, Generic)
   deriving Show via Quiet EpochSize
-  deriving newtype (Enum, Num, Real, Integral, ToCBOR, FromCBOR, NoThunks, ToJSON, FromJSON, NFData)
+  deriving newtype (Enum, ToCBOR, FromCBOR, NoThunks, ToJSON, FromJSON, NFData)
+
+newtype EpochInterval where
+  EpochInterval :: {unEpochInterval :: Word32} -> EpochInterval
```

Comment:
Out of curiosity, why the GADT syntax?

Reply:
Ha, thanks for the catch! This makes me curious as well as I did not intend to do this. Here is what I think might have happened: when I copied `EpochInterval`, I realised it has `EncCBOR`, `DecCBOR` instances and immediately applied a code action via `hls` in order to attempt to import those. That did not happen obviously since those are defined in `cardano-ledger`, however the code action formatted the code and also transformed it into a `GADT` which I didn't realise. I'll remove it as it is unnecessary. 

By the way, regarding the `EncCBOR`, `DecCBOR` instances: I guess we would prefer to provide those instance definitions as orphans in `cardano-ledger` rather than moving the whole class declarations here and deriving them inplace, right?

Reply:
> code action formatted the code and also transformed it into a GADT

That's why I stay away from too much smartness in my tooling :smile: 

> By the way, regarding the EncCBOR, DecCBOR instances: I guess we would prefer to provide those instance definitions as orphans in cardano-ledger rather than moving the whole class declarations here and deriving them inplace, right?

Those instances will need to be defined right next to `EncCBOR`/`DecBOR` type classes in `cardano-ledger-binary`, which will ensure that they are not orphans